.github/workflows/verify.yml

```diff
@@ -80,42 +90,30 @@ jobs:
           key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
           restore-keys: |
             ${{ runner.os }}-go-
-
       - name: Run integration tests
-        run: bash integration_tests/run-all.sh
-
+        shell: bash
+        run: integration_tests/run-all.sh
   lint:
-    name: Lint  # TODO remove (requires repo settings update)
+    if: needs.changes.outputs.go == 'true'
+    name: Check linter issues with golangci-lint tool
     runs-on: ubuntu-latest
+    needs:
+      - changes
     steps:
-
-      - uses: actions/checkout@v2
-
+      - uses: actions/checkout@v3
       - name: golangci-lint
         uses: golangci/golangci-lint-action@v2
         with:
-          version: v1.43
           args: --timeout 5m0s
-
-  license:
-    name: License  # TODO remove (requires repo settings update)
+  check:
+    if: always()
+    needs:
+      - unit-tests
```

Comment:
I think we need to apply the same approach then for other check: lint.yml, spell.yml, on_demand_molecule.yml, etc.
BTW superlint checks appeared to be running surprisingly long (~45 minutes). We had to disable them now, since needed to issue the release. I think we will need to optimise the super lint checks to make it much faster.

Reply:
> I think we need to apply the same approach then for other check: lint.yml, spell.yml, on_demand_molecule.yml, etc. 

Do you mean, to put them into a single workflow file? It could be possible, but the disadvantage will be to re-run a single failure job or disable a specific workflow.

> BTW superlint checks appeared to be running surprisingly long (~45 minutes). We had to disable them now, since needed to issue the release. I think we will need to optimise the super lint checks to make it much faster.

Well, the super-linter contains a collection of linters, which combine with the size of the project as well as the variety of languages used results in longer execution. Another alternative could be to change it to run it once a week at least, in that way we can monitor any issue.

Reply:
> Do you mean, to put them into a single workflow file?

I'm thinking here from the branch protection rules point of view again. It would be great to require running for example spell.yml if there are any changes in doc files, so that branch protection rules will not allow to merge a PR with failing spell checks. 
As for how to do this, put everything into a single workflow file and use all-greens action is one of the options for sure. Another option is to enable spell checks in the protection rules, but as it contains filters (paths) on the pull_request level, we should probably apply the same refactoring there as in this PR.


> Another alternative could be to change it to run it once a week at least, in that way we can monitor any issue.

It can be a reasonable option.